packages/pubsub/src/topic.js

```diff
@@ -224,39 +92,186 @@ function Topic(pubsub, name) {
   this.iam = new IAM(pubsub, this.name);
 }
 
-util.inherits(Topic, commonGrpc.ServiceObject);
-
 /**
- * Format a message object as the upstream API expects it.
+ * Format the name of a topic. A Topic's full name is in the format of
+ * 'projects/{projectId}/topics/{topicName}'.
  *
  * @private
  *
- * @return {object}
+ * @return {string}
+ */
+Topic.formatName_ = function(projectId, name) {
+  // Simple check if the name is already formatted.
+  if (name.indexOf('/') > -1) {
+    return name;
+  }
+  return 'projects/' + projectId + '/topics/' + name;
+};
+
+/**
+ * Create a topic.
+ *
+ * @param {object=} gaxOpts - Request configuration options, outlined
+ *     here: https://googleapis.github.io/gax-nodejs/CallSettings.html.
+ * @param {function=} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {module:pubsub/topic} callback.topic - The topic.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * topic.create(function(err, topic, apiResponse) {
+ *   if (!err) {
+ *     // The topic was created successfully.
+ *   }
+ * });
+ *
+ * //-
+ * // If the callback is omitted, we'll return a Promise.
+ * //-
+ * topic.create().then(function(data) {
+ *   var topic = data[0];
+ *   var apiResponse = data[1];
+ * });
+ */
+Topic.prototype.create = function(gaxOpts, callback) {
+  this.pubsub.createTopic(this.name, gaxOpts, callback);
+};
+
+/**
+ * Create a subscription to this topic.
+ *
+ * @resource [Subscriptions: create API Documentation]{@link https://cloud.google.com/pubsub/docs/reference/rest/v1/projects.subscriptions/create}
+ *
+ * @throws {Error} If subscription name is omitted.
+ *
+ * @param {string=} name - The name of the subscription.
+ * @param {object=} options - See a
+ *     [Subscription resource](https://cloud.google.com/pubsub/docs/reference/rest/v1/projects.subscriptions)
+ * @param {number} options.ackDeadline - The maximum time after receiving a
+ *     message that you must ack a message before it is redelivered.
+ * @param {object} options.flowControl - Flow control configurations for
+ *     receiving messages.
+ * @param {number} options.flowControl.maxBytes - The maximum number of bytes
+ *     in un-acked messages to allow before the subscription pauses incoming
+ *     messages. Defaults to 20% of free memory.
+ * @param {number} options.flowControl.maxMessages - The maximum number of
+ *     un-acked messages to allow before the subscription pauses incoming
+ *     messages. Default: Infinity.
```

Comment:
As I mentioned above, it's good to be clear that these are not persistent config options, but are rather properties of opening a subscription in the client